### PR TITLE
Improved compatibility of openlayers DrawSupport between all the invo…

### PR DESCRIPTION
**Pull request by mbarto**
…lved functionalities: QueryBuilder, FeatureEditor, Annotations

## Description
See title

## Issues
 - See title

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
DrawSupport behaviour is not consistent with pre / annotations work for other tools

**What is the new behavior?**
DrawSupport behaviour is consistent with pre / annotations work for other tools

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/map/openlayers/DrawSupport.jsx [hunk 1]
```diff
@@ -63,7 +61,8 @@ class DrawSupport extends React.Component {
         onChangeDrawingStatus: () => {},
         onGeometryChanged: () => {},
         onDrawStopped: () => {},
-        onEndDrawing: () => {}
+        onEndDrawing: () => {},
+        onTextChanged: () => {}
     };
```
Comment (MV88) [inline web/client/components/map/openlayers/DrawSupport.jsx]: is this used somewhere?

**Comment by MV88** [comment]
ASAIU Your proposal is to use some defaults style for vector layers and others for annotations?

**Comment by mbarto** [comment]
My proposal is that each module should be able to use its own preferred style, with a proper default style if no one is specified by the module (owner)

**Comment by MV88** [comment]
This PR is an improvement of the [2612 PR](https://github.com/geosolutions-it/MapStore2/pull/2612)
it is linked to the same solved issues
 - Fix #2529
 - Fix #2531
 - Fix #2532

**Pull request merged**